lib/ui/painting.dart

```diff
@@ -1567,6 +1567,22 @@ class FrameInfo extends NativeFieldWrapperClass2 {
   Image get image native 'FrameInfo_image';
 }
 
+/// The default max amount of memory to use for caching decoded animated image
+/// frames compared to total undecoded size. See [decodedCacheRatioCapOverride].
+const double kDefaultDecodedCacheRatioCap = 25.0;
+
+/// By default individual frames of animated images are cached into memory to
+/// avoid using CPU to re-decode them for every loop in the animation. This
+/// behavior will result in out of memory crashes when decoding large
+/// (or large amounts of) animated images. Set this value to limit how much
+/// memory each animated image is allowed to use to cache decoded frames
+/// compared to its compressed size. For example, setting this to `2.0` means
+/// that a 400KB GIF would be allowed at most to use 800KB of memory caching
+/// unessential decoded frames. A setting of `1.0` or less disables all caching
+/// of unessential decoded frames. See [kDefaultDecodedCacheRatioCap] for the
+/// default value.
+double decodedCacheRatioCapOverride;
```

Comment:
There's no setter here, meaning that setting the value has no effect. Document how to make it take effect (e.g. "this value only takes effect when `instantiateImageCodec` is invoked; to change the cache on the fly, the image must be redecoded with a new value).

But also, this is a global value with non-immediate side-effects, which is generally frowned upon. I would recommend making this an argument to `instantiateImageCodec` rather than a global variable.

Reply:
(Consider how a framework that wanted to pick the ratio on a per-image basis would have to be written. Or what would happen if two libraries with different opinions were to interact. They'd be clobbering each other's setting of this variable.)

Reply:
On it being a global: making it a param of the constructor forces this to bubble out pretty far in order for users to actually set it when constructing their images, since `Codec` isn't being constructed directly and ends up being created from several different top level widgets. From what I can tell `AssetImage`, `ExtractAssetImage`, `FileImage`, `NetworkImage`, and `MemoryImage` all end up eventually instantiating `Codec` and would need the param added to their constructors too in order to pass it through. There also looks like there's static constructors attached to `Image` itself for each of those classes: `Image.asset`, `Image.file`, `Image.network`, `Image.memory`. I'm good with adding this new param throughout if you think it's preferable to a global setter? When I first wrote this I was thinking it was a bigger problem to have a param existing across all those signatures in order to flip a setting way over in this class, but I hadn't thought about people wanting to potentially set it on a per-image basis.

Reply:
Talked offline. It's possible to avoid both of these problems if the multiple `instantiateImageCodec` calls in `image_provider.dart` are diverted into a new singular method in `binding.dart`. A global is more of an accepted practice on the framework side, so the default can be defined and overridden there.